config/raw_config.go

```diff
@@ -46,6 +48,19 @@ func NewRawConfig(raw map[string]interface{}) (*RawConfig, error) {
 	return result, nil
 }
 
+// Copy returns a copy of this RawConfig, uninterpolated.
+func (r *RawConfig) Copy() *RawConfig {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	result, err := NewRawConfig(r.Raw)
```

Comment:
I assume this is fine, but double checking is always good. It is okay that we don't copy `Key`?